src/main/java/org/wildfly/swarm/plugin/repository/RepositoryBuilderMojo.java

```diff
@@ -99,17 +99,38 @@ private void santizeRepo(Path repoDirPath) throws IOException {
         getLog().info("Remove unneeded files from local M2 Repo, " + repoDirPath.toAbsolutePath());
 
         Files.walkFileTree(repoDirPath, new SimpleFileVisitor<Path>() {
+            private boolean pruneDirectory = false;
+
             @Override
             public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
-                if (file.endsWith("_remote.repositories")) {
+                // If we need to prune communtiy artifacts, ie those from Central, then remove from repository
+                if (!pruneDirectory && Boolean.parseBoolean(removeCommunity)) {
+                    if (!file.toString().contains("redhat-")) {
```

Comment:
This assumes that either all files in a directory contain `redhat-`, or none of them do. (With the exception of `_remote.repositories` and `*.lastUpdated`.) Otherwise there could be a single file that _woultn't_ contain `redhat-`, that could be first in iteration order, wouldn't be deleted, but then there would be a file that _would_ contain `redhat`, which would set the flag, and afterwards the `postVisitDirectory` would try to delete the directory and fail.

I don't know if this assumption is correct, but probably yes. Just wanted to bring this up.